src/TagsInput/TagsInput.js

```diff
@@ -0,0 +1,162 @@
+  +placeholder: string,
+  +selected?: string[],
+  +disabled?: boolean,
+  +fontSize: number,
+  +onClearPress?: () => void,
+  +value?: string,
+|};
+
+type State = {
+  value: string,
+  containerWidth: ?number,
+  inputWidth: ?number,
+};
+
+export default class TagInput extends React.Component<Props, State> {
+  inputRef: ?{ current: null | React.Element<typeof TextInput> };
+  static defaultProps = {
+    fontSize: 16,
+  };
+
+  constructor(props: Props) {
+    super(props);
+    this.inputRef = React.createRef();
+
+    this.state = {
+      value: props.value ?? '',
+      containerWidth: null,
+      inputWidth: null,
+    };
+  }
+
+  componentDidUpdate = (prevProps: Props) => {
+    const { value } = this.props;
+    if (value !== prevProps.value) {
+      this.setState({ value });
+    }
+  };
+
+  setContainerWidth = (event: ViewLayoutEvent) => {
+    this.setState({ containerWidth: event.nativeEvent.layout.width });
+  };
+
+  setInputWidth = (event: ViewLayoutEvent) => {
+    this.setState({ inputWidth: event.nativeEvent.layout.width });
+  };
+
+  getWidth = (width: ?number) => {
+    const { containerWidth } = this.state;
+    if (containerWidth && width) {
+      const newWidth = containerWidth - width;
+      return newWidth < 0 ? 'auto' : newWidth;
+    }
+    return null;
+  };
+
+  getPlaceholder = () => {
+    const { selected, placeholder } = this.props;
+    return selected?.length === 0 && placeholder ? placeholder : null;
+  };
+
+  handleChange = (value: string) => {
+    const { value: oldValue } = this.state;
+    const { onChangeText } = this.props;
+    if (value !== oldValue) {
+      this.setState({ value });
+      onChangeText(value);
+    }
+  };
+
+  handleClear = () => {
+    const { onClearPress } = this.props;
+    this.setState({ value: '' });
+
+    // $FlowFixMe property focus is missing in object type
+    Platform.OS === 'web' && this.inputRef.current.focus();
+
+    if (onClearPress) {
+      onClearPress();
+    }
+  };
+
+  render() {
+    const { selected, fontSize, label, disabled } = this.props;
+    const { value, inputWidth } = this.state;
+    const isButtonDisabled = !value.length ?? disabled;
+
+    const dynamicStyle = StyleSheet.create({
+      deleteButton: {
+        opacity: value.length > 0 ? 1 : 0,
+      },
+      label: {
+        fontSize,
+      },
+    });
+
+    return (
+      <View style={styles.container}>
+        <Text weight="bold" style={(styles.label, dynamicStyle.label)}>
+          {label}
+        </Text>
+        <View style={styles.fieldContainer} onLayout={this.setContainerWidth}>
+          <TagsContainer
+            maxWidth={this.getWidth(inputWidth)}
+            tags={selected}
+            fontSize={fontSize}
+          />
+          <InputField
+            ref={this.inputRef}
+            fontSize={fontSize}
+            disabled={disabled}
+            minWidth={MIN_WIDTH}
+            maxWidth={this.getWidth(MIN_WIDTH)}
```

Comment:
Maybe the naming could be improved here: why is `maxWidth` linked to `MIN_WIDTH`?

Reply:
Because of in the getWidth method I subtract passing param, in this case, MIN_WIDTH from the width of the whole container.
And because TagsContainer minWidth is also MIN_WIDTH, maxWidth of the InputField must respect this. 

I will try to improve it to something more readable.